launcher/QObjectPtr.h

```diff
@@ -1,91 +1,48 @@
 #pragma once
 
+#include <QObject>
+#include <QSharedPointer>
 #include <functional>
 #include <memory>
-#include <QObject>
 
-namespace details
-{
-struct DeleteQObjectLater
+namespace details {
+[[maybe_unused]] static void do_delete_later(QObject* obj)
 {
-    void operator()(QObject *obj) const
-    {
+    if (obj)
         obj->deleteLater();
-    }
-};
 }
+struct DeleteQObjectLater {
+    void operator()(QObject* obj) const { do_delete_later(obj); }
+};
+
+}  // namespace details
+
 /**
  * A unique pointer class with unique pointer semantics intended for derivates of QObject
  * Calls deleteLater() instead of destroying the contained object immediately
  */
-template<typename T> using unique_qobject_ptr = std::unique_ptr<T, details::DeleteQObjectLater>;
+template <typename T>
+using unique_qobject_ptr = std::unique_ptr<T, details::DeleteQObjectLater>;
```

Comment:
We can use `QScopedPointer here`: https://doc.qt.io/qt-6/qscopedpointer.html

Reply:
there is a premade struct called `QScopedPointerDeleteLater` that seems to have.... about no docs that can be used in the same way as the "DeleteQObjectLater" here

Reply:
```suggestion
using unique_qobject_ptr = QScopedPointer<T, QScopedPointerDeleteLater>;
```

Reply:
this will also make the whole details:: namespace useless (after applying the other suggestion from scrumplex)

Reply:
NOTE : from the code of QScopedPointerDeleteLater it does do the nullcheck